x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Verifier.java

```diff
@@ -640,44 +655,99 @@ private static void checkRemoteEnrich(LogicalPlan plan, Set<Failure> failures) {
      * early in the execution, rather than fail at the compute engine level.
      * In the future we will be able to handle MATCH at the compute and we will no longer need these checks.
      */
-    private static void checkFilterMatchConditions(LogicalPlan plan, Set<Failure> failures) {
+    private static void checkMatchPredicateArgumentTypes(LogicalPlan plan, Set<Failure> failures) {
         if (plan instanceof Filter f) {
             Expression condition = f.condition();
 
-            Holder<Boolean> hasMatch = new Holder<>(false);
             condition.forEachDown(MatchQueryPredicate.class, mqp -> {
-                hasMatch.set(true);
                 var field = mqp.field();
                 if (field instanceof FieldAttribute == false) {
-                    failures.add(fail(mqp, "MATCH requires a mapped index field, found [" + field.sourceText() + "]"));
+                    failures.add(fail(mqp, "[:] operator requires a mapped index field, found [" + field.sourceText() + "]"));
                 }
 
                 if (DataType.isString(field.dataType()) == false) {
                     var message = LoggerMessageFormat.format(
                         null,
-                        "MATCH requires a text or keyword field, but [{}] has type [{}]",
+                        "[:] operator requires a text or keyword field, but [{}] has type [{}]",
                         field.sourceText(),
                         field.dataType().esType()
                     );
                     failures.add(fail(mqp, message));
                 }
             });
-
-            if (canPushToSource(condition, x -> false)) {
-                return;
-            }
-            if (hasMatch.get()) {
-                failures.add(fail(condition, "Invalid condition using MATCH"));
-            }
         }
     }
 
+    /**
```

Comment:
I've slightly refactored and documented the following methods so they can be reused for Match function and Match predicates